app/controllers/recipe_ingredients_controller.rb

```diff
@@ -0,0 +1,18 @@
+class RecipeIngredientsController < ApplicationController
+    def new
+        @recipe = Recipe.find(params[:id])
+    end
+
+    def create
+        recipe = Recipe.find(params[:id])
+        recipe.ingredients.create(ingredient_params)
+
+        redirect_to "/recipes/#{recipe.id}"
+    end
+
+    private
+
+    def ingredient_params
+        params.permit(:name, :cost)
+    end
+end
```

Comment:
This quite a bit going on in this controller that doesn't completely align with the user story. The intent is not to create a new ingredient but to associate an already existing ingredient to a recipe. Your controller action should look something like this:

```ruby
def create
  recipe = Recipe.find(params[:id]) # :id comes from the route (ie. "POST /routes/:id/ingredients"
  ingredient = Ingredient.find(params[:ingredient_id]) # :ingredient_id should come from the form

  RecipeIngredient.create!(recipe: recipe, ingredient: ingredient) # We create a new entry in the join table

  redirect_to "/recipes/#{recipe.id}"
end
```

Having the `ingredient_params` method is ok if you want to do strong params, but the `new` action is not necessary to satisfy the user story.